pkg/kubectl/describe.go

```diff
@@ -1395,6 +1396,32 @@ func DescribeEvents(el *api.EventList, w io.Writer) {
 	}
 }
 
+// DeploymentDescriber generates information about a deployment.
+type DeploymentDescriber struct {
+	client.Interface
+}
+
+func (dd *DeploymentDescriber) Describe(namespace, name string) (string, error) {
+	d, err := dd.Experimental().Deployments(namespace).Get(name)
```

Comment:
The comparison I'm drawing is to the pod describer where we fetch the replication controllers that overlap the pod.  I think users would expect the same here - I'm ok with showing status, but it is common to want to know whether you have multiple RCs in flight during a deployment.  Getting the list of RCs and even showing what the D controller thinks is the status is both good.  The note about not failing was just that if the RC is borked for some reason we don't want to prevent describe from working.